src/EditorFeatures/Core/Shared/Utilities/ThreadingContext.cs

```diff
@@ -49,5 +55,37 @@ public JoinableTaskFactory JoinableTaskFactory
         {
             get;
         }
+
+        public JoinableTaskCollection ShutdownBlockingTasks { get; }
+
+        private JoinableTaskFactory ShutdownBlockingTaskFactory { get; }
+
+        public CancellationToken DisposalToken => _disposalTokenSource.Token;
+
+        public JoinableTask RunWithShutdownBlockAsync(Func<CancellationToken, Task> func)
+        {
+            return ShutdownBlockingTaskFactory.RunAsync(() => func(DisposalToken));
+        }
+
+        public void Dispose()
```

Comment:
who ends up calling this?

Reply:
The MEF container calls this when it is disposed.